crates/router/src/connector/utils.rs

```diff
@@ -166,14 +170,32 @@ impl PaymentsAuthorizeRequestData for types::PaymentsAuthorizeData {
     }
 }
 
+pub trait PaymentsCompleteAuthorizeRequestData {
+    fn is_auto_capture(&self) -> Result<bool, Error>;
+}
+
+impl PaymentsCompleteAuthorizeRequestData for types::CompleteAuthorizeData {
```

Comment:
```suggestion
impl types::CompleteAuthorizeData {
```

we don't require `PaymentsCompleteAuthorizeRequestData` trait